src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -86,6 +87,69 @@
 )
 
 
+@postgres_dialect.segment(replace=True)
+class ArrayLiteralSegment(BaseSegment):
+    """An array literal segment."""
+
+    type = "array_literal"
+    match_grammar = Sequence(
+        "ARRAY",
+        Bracketed(
+            Delimited(Ref("LiteralGrammar"), optional=False),
+            bracket_type="square",
+        ),
+    )
+
+
+postgres_dialect.replace(
```

Comment:
Rather than copying and modifying the source code from the ANSI `BaseExpressionGrammar`, could you look at extending it as seen in the BigQuery dialect [here](https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/dialects/dialect_bigquery.py#L112L114)? This technique is more maintainable, in that changes to the ANSI base will be automatically reflected in Postgres.

Reply:
ah, sure thing, makes sense.